core/src/main/java/nl/nn/adapterframework/jdbc/dbms/GenericDbmsSupport.java

```diff
@@ -286,85 +284,69 @@ protected boolean doIsTablePresent(Connection conn, String tablesTable, String s
-				log.debug("execute statement");
-				ResultSet rs = stmt.executeQuery();
-				log.debug("statement executed");
 				rs.close();
-				return true;
-			} catch (SQLException e) {
-				if (log.isDebugEnabled()) log.debug("exception checking for existence of ["+tableName+"] using query ["+query+"]", e);
-				return false;
-			} finally {
-				if (stmt!=null) {
-					stmt.close();
-				}
+			} catch (Exception e) {
+				if (log.isDebugEnabled()) log.debug("there was an error closing the resulset", e);
 			}
 		}
-		catch(SQLException e) {
-			throw new JdbcException(e);
-		}
+	    return tExists;
 	}
 
-	@Override
-	public boolean isColumnPresent(Connection conn, String tableName, String columnName) throws SQLException {
-		PreparedStatement stmt = null;
-		String query=null;
+	
+	private boolean isColumnPresent(Connection conn, String schemaName, String tableName, String columnName) throws SQLException {
+		ResultSet rs = null;
+		boolean cExists = Boolean.FALSE;
 		try {
-			query = "SELECT count(" + columnName + ") FROM " + tableName;
-			stmt = conn.prepareStatement(query);
-
-			ResultSet rs = null;
+			rs = conn.getMetaData().getColumns(null, schemaName, tableName, columnName);
+			cExists = !rs.isAfterLast();
+		}catch(SQLException e) {
+			log.warn("exception checking for existence of column ["+columnName+"] in table ["+tableName+"]", e);
+		}finally {
 			try {
-				rs = stmt.executeQuery();
-				return true;
-			} catch (SQLException e) {
-				if (log.isDebugEnabled()) log.debug("exception checking for existence of column ["+columnName+"] in table ["+tableName+"] executing query ["+query+"]", e);
-				return false;
-			} finally {
-				if (rs != null) {
-					rs.close();
-				}
-			}
-		} catch (SQLException e) {
-			log.warn("exception checking for existence of column ["+columnName+"] in table ["+tableName+"] preparing query ["+query+"]", e);
-			return false;
-		} finally {
-			if (stmt != null) {
-				stmt.close();
+				rs.close();
+			} catch (Exception e) {
+				if (log.isDebugEnabled()) log.debug("there was an error closing the resulset", e);
 			}
 		}
+		return cExists;
+	}
+
+	@Override
+	public boolean isColumnPresent(Connection conn, String tableName, String columnName) throws SQLException {
+		return this.isColumnPresent(conn, null, tableName, columnName);
 	}
 
 	protected boolean doIsTableColumnPresent(Connection conn, String columnsTable, String schemaColumn, String tableNameColumn, String columnNameColumn, String schemaName, String tableName, String columnName) throws JdbcException {
-		String query="select count(*) from "+columnsTable+" where upper("+tableNameColumn+")=? and upper("+columnNameColumn+")=?";
+		boolean isPresent = Boolean.FALSE;
 		if (StringUtils.isNotEmpty(schemaName)) {
 			if (StringUtils.isNotEmpty(schemaColumn)) {
-				query+=" and upper("+schemaColumn+")='"+schemaName.toUpperCase()+"'";
+				try {
+					isPresent = this.isColumnPresent(conn, schemaColumn, tableName, columnName);
```

Comment:
doIsTableColumnPresent() should not call isColumnPresent(), because it is meant to provide a way to override isColumnPresent() in descender classes, like OracleDbmsSupport, H2DbmsSupport, MsSqlServerDbmsSupport.
In those classes it is currently used to override isColumntPresent().
Please:
- remove the implementations of doIsTableColumnPresent in OracleDbmsSupport, H2DbmsSupport and MsSqlServerDbmsSupport, to enable that they use the standard Jdbc solution.
- implement doIsTableColumnPresent with the same queries, but then with placeholders ('?') and parameters instead of
literals.

Please run also iaf-test to see that your code runs correctly in a Frank application